float64[] q
float64[] qd
float64[] uGP
---
